Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -109,6 +109,75 @@ namespace AZ
             return m_wasPreFinalized;
         }
 
+        template<typename T>
```

Comment:
My eyes might be tired, but why does this need to be a template function if the returned type is also a variant? Internally it looks like we're chaining the `T` cast right after the rtti-based cast but does the second cast do anything? (Probably should just be referring to `MaterialPropertyValue` to understand this better, but just noting that this is an open question as the reader for the time being)

Reply:
MaterialPropertyValue is a AZStd::variant, which does not support any kind of casting. If it's an int, the only valid call to variant::GetValue is GetValue<int>(). This function tries to convert the given MaterialPropertyValue to a different type of MaterialPropertyValue, according to the template parameter T.